### DISCUSSION TITLE
how to logging response body into access.log

### DISCUSSION THREAD

**Discussion by gracekk** [apache/apisix#7297] [Q&A] [answered]
I try to use $resp_body in config.yml but failed. 
Any ideas?
Or Can I use http-logger plugin write resp_body into file?

nginx_config:
  http:
    error_log_level: "info"
    enable_access_log: true
    access_log: logs/access.log
    access_log_format: '{"@timestamp":"$time_iso8601","host":"$hostname","server_ip":"$server_addr","client_ip":"$remote_addr","xff":"$http_x_forwarded_for","domain":"$host","url":"$uri","referer":"$http_referer","args":"$args","upstreamtime":"$upstream_response_time","responsetime":"$request_time","request_method":"$request_method","status":"$status","size":"$body_bytes_sent","resp_body":"$resp_body","request_body":"$request_body","request_length":"$request_length","protocol":"$server_protocol","upstreamhost":"$upstream_addr","file_dir":"$uri","http_user_agent":"$http_user_agent"}'
    access_log_format_escape: json
  http_configuration_snippet: |
    log_format  my-custom-logs-format
    '{"@timestamp":"$time_iso8601",'
    '"host":"$hostname",'
    '"server_ip":"$server_addr",'
    '"client_ip":"$remote_addr",'
    '"xff":"$http_x_forwarded_for",'
    '"domain":"$host",'
    '"url":"$uri",'
    '"referer":"$http_referer",'
    '"args":"$args",'
    '"upstreamtime":"$upstream_response_time",'
    '"responsetime":"$request_time",'
    '"request_method":"$request_method",'
    '"status":"$status",'
    '"size":"$body_bytes_sent",'
    '"request_body":"$request_body",'
    '"request_length":"$request_length",'
    '"protocol":"$server_protocol",'
    '"upstreamhost":"$upstream_addr",'
    '"file_dir":"$uri",'
    '"resp_body:"$resp_body",'
    '"http_user_agent":"$http_user_agent"'
    '}';

**Comment by soulbird** [comment, marked as answer]
There is no built-in variable support to get the response body, if your response body size is unpredictable, I suggest you don't log it, it will make your log size very large. 
But, If you insist on doing this, you can use the following way:
1、Set a var like: 
```
set $resp_body '';
```
2、set log format:
```
access_log_format $resp_body;
```
3、Customize a plugin to assign a value to $resp_bdoy：
```lua
function _M.body_filter(conf, ctx)
    local body = core.response.hold_body_chunk(ctx)
    ctx.var.resp_body = body
    .............
end
```

**Reply by gracekk** [reply]
Thx for replay.
set $resp_body '' in this part?
http_server_configuration_snippet: |
       set $resp_body '';

**Comment by juzn01** [comment]
request feature: can you add this as an plugin like proxy-mirror ???,some time my develop team need response from upstream via apisix before apisix send backto client.
